pymeasure/instruments/hp/hp3478A.py

```diff
@@ -463,3 +463,88 @@ def trigger(self):
     def trigger(self, value):
         trig_set = self.TRIGGERS[strict_discrete_set(value, self.TRIGGERS)]
         self.write(trig_set)
+
+    @property
+    def calibration_data(self):
+        """Return the full calibration data as an array of 256 values between 0 and 15.
+
+        The calibration data of an HP 3478A is stored in a 256x4 SRAM that is
+        permanently powered by a 3v Lithium battery. When the battery runs
+        out, the calibration data is lost, and recalibration is required.
+
+        This property fetches the calibration data so that it can be backed up.
+
+        """
+        cal_data = []
+        for addr in range(0, 256):
+            # To fetch one nibble: 'W<address>', where address is a raw 8-bit number.
+            cmd = bytes([ord('W'), addr])
+            self.write_bytes(cmd)
+            rvalue = self.read_bytes(1)[0]
+            if rvalue < 64 or rvalue >= 80:
+                raise Exception("calibration nibble out of range")
+            cal_data.append(rvalue-64)
```

Comment:
I first got confused, why do you check the range (64, 80) if values are in the range (0,15). A comment could help.
```suggestion
            # rvalue is a hexadecimal ASCII character.
            if rvalue < 64 or rvalue >= 80:
                raise Exception("calibration nibble out of range")
            cal_data.append(rvalue-64)
```